pkg/tests/reward_test.go

```diff
@@ -0,0 +1,233 @@
+		OutputID:        accountOutput.OutputID(),
+		ExpirySlot:      blockIssuerFeatExpirySlot,
+		BlockIssuerKeys: iotago.BlockIssuerKeys{blockIssuerFeatKey},
+		StakeEndEpoch:   0,
+		ValidatorStake:  0,
+	}, ts.Nodes()...)
+
+	ts.AssertAccountDiff(accountID, block2Slot, &model.AccountDiff{
+		BICChange:              -iotago.BlockIssuanceCredits(0),
+		PreviousUpdatedSlot:    0,
+		NewExpirySlot:          blockIssuerFeatExpirySlot,
+		PreviousExpirySlot:     blockIssuerFeatExpirySlot,
+		NewOutputID:            accountOutput.OutputID(),
+		PreviousOutputID:       ts.DefaultWallet().Output("TX1:0").OutputID(),
+		BlockIssuerKeysAdded:   iotago.NewBlockIssuerKeys(),
+		BlockIssuerKeysRemoved: iotago.NewBlockIssuerKeys(),
+		ValidatorStakeChange:   -int64(stakedAmount),
+		StakeEndEpochChange:    -int64(endEpoch),
+		FixedCostChange:        -int64(fixedCost),
+		DelegationStakeChange:  0,
+	}, false, ts.Nodes()...)
+}
+
+func Test_RewardInputCannotPointToNFTOutput(t *testing.T) {
+	ts, node1, node2 := setupDelegationTestsuite(t)
+	defer ts.Shutdown()
+
+	// CREATE NFT FROM BASIC UTXO
+	var block1Slot iotago.SlotIndex = 1
+	ts.SetCurrentSlot(block1Slot)
+
+	input := ts.DefaultWallet().Output("Genesis:0")
+	nftOutput := builder.NewNFTOutputBuilder(ts.DefaultWallet().Address(), input.BaseTokenAmount()).MustBuild()
+	tx1 := ts.DefaultWallet().CreateSignedTransactionWithOptions(
+		"TX1",
+		mock.WithInputs(utxoledger.Outputs{input}),
+		mock.WithOutputs(iotago.Outputs[iotago.Output]{nftOutput}),
+		mock.WithAllotAllManaToAccount(ts.CurrentSlot(), ts.DefaultWallet().BlockIssuer.AccountID),
+	)
+
+	block1 := ts.IssueBasicBlockWithOptions("block1", ts.DefaultWallet(), tx1)
+
+	latestParents := ts.CommitUntilSlot(block1Slot, block1.ID())
+
+	ts.AssertTransactionsExist([]*iotago.Transaction{tx1.Transaction}, true, node1, node2)
+	ts.AssertTransactionsInCacheAccepted([]*iotago.Transaction{tx1.Transaction}, true, node1, node2)
+
+	// ATTEMPT TO POINT REWARD INPUT TO AN NFT OUTPUT
+	inputNFT := ts.DefaultWallet().Output("TX1:0")
+	prevNFT := inputNFT.Output().Clone().(*iotago.NFTOutput)
+	nftOutput = builder.NewNFTOutputBuilderFromPrevious(prevNFT).NFTID(iotago.NFTIDFromOutputID(inputNFT.OutputID())).MustBuild()
+
+	tx2 := ts.DefaultWallet().CreateSignedTransactionWithOptions(
+		"TX2",
+		mock.WithInputs(utxoledger.Outputs{inputNFT}),
+		mock.WithRewardInput(
+			&iotago.RewardInput{Index: 0},
+			0,
+		),
+		mock.WithCommitmentInput(&iotago.CommitmentInput{
+			CommitmentID: ts.DefaultWallet().Node.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment().Commitment().MustID(),
+		}),
+		mock.WithOutputs(iotago.Outputs[iotago.Output]{nftOutput}),
+		mock.WithAllotAllManaToAccount(ts.CurrentSlot(), ts.DefaultWallet().BlockIssuer.AccountID),
+	)
+
+	ts.IssueBasicBlockWithOptions("block2", ts.DefaultWallet(), tx2, mock.WithStrongParents(latestParents...))
+
+	ts.Wait(node1, node2)
+
+	// TODO: Assertions do not pass for node2 - why?
```

Comment:
This one is probably worth figuring out, anything interesting in the logs?

Reply:
We can leave this as is for now and add a TODO here referencing this issue https://github.com/iotaledger/iota-core/issues/580
And we check for failure at all nodes once that issue is closed rather than just node1